src/main/java/sirius/tagliatelle/emitter/Emitter.java

```diff
@@ -101,4 +105,36 @@ public Position getStartOfBlock() {
      *                improved error messages).
      */
     public abstract void visitExpressions(@Nonnull Function<Position, ExpressionVisitor> visitor);
+
+    /**
+     * Emits escaped comments to the rendering context with {@link GlobalRenderContext.DebugLevel#DEBUG} level.
+     *
+     * @param context the render context which provides access to the local and global environment and also the target
+     *                for the generated text
+     * @param message text to output with {@link GlobalRenderContext.DebugLevel#DEBUG}
+     */
+    public void emitDebugMessage(@Nonnull LocalRenderContext context, String message) {
```

Comment:
maybe convert this to a pattern like Strings.apply -> message, Object... args and only construct the effective message if the level matches - this should save us from tons of unnecessary string concats und toString calls.

Reply:
`emitDebugMessage` and `emitTraceMessage` converted to accept an Object... arguments.
In turn, `emitMessage` first validates if comments shall be emitted before using the message and arguments with `Strings.apply()`